### PR TITLE
[WIP] Migrate properties to fileid

### LINKED ISSUE #25563: oc_properties must use fileid instead of path
The oc_properties table contains any custom Webdav properties that might have been set by third party clients.

The trouble is that this table contains the full path to the target file and doesn't properly detect renames... so properties can get lost.
Furthermore, the column is too small and longer paths might not fit.

To be able to better preserve properties, we should make this table use the fileid instead of the path.

@DeepDiver1975

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
Use migrations for all dav tables.
Use fileid instead of path in dav properties

## Related Issue
Fixes https://github.com/owncloud/core/issues/25563

## Motivation and Context
t this table contains the full path to the target file and doesn't properly detect renames... so properties can get lost.
Furthermore, the column is too small and longer paths might not fit.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/dav/appinfo/Migrations/Version20170116170538.php [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+			]);
+			$table->setPrimaryKey(['id']);
+			$table->addIndex(['file_id'], 'fileid_index');
+		} else {
+	       // Update from 9.1
+			// TODO: Introduce file_id
+			// TODO: drop userid and propertypath
+		}
+
+		if (!$schema->hasTable("${prefix}addressbooks")) {
```
Comment (PVince81) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: oh, do we really also need these tables ? are they related to oc_properties ?

File: apps/dav/appinfo/Migrations/Version20170116170538.php [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+
+    }
+
+    /**
+     * @param Schema $schema
+     */
+    public function down(Schema $schema)
+    {
+		// We can't migrate below 10.0, can we?
```
Comment (PVince81) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: no, let's always ignore the `down` part for now...

File: apps/dav/appinfo/Migrations/Version20170116170538.php (hunk 1 above)
Comment (VicDeo) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: @PVince81 oc_properties  table belongs to `dav` app and is contained in core db schema by mistake now. 
To use migrations on dav app all its schema needs to be ported to migrations first.
This is how I see it.

Comment (PVince81) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: I'm not sure if this is necessary. But maybe it's also ok to have it now, @DeepDiver1975 what do you think ?

@VicDeo make sure the DB dump diffs before and after are the same.

Comment (VicDeo) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: @PVince81 ok, Let's assume we don't need all this tables. This table is in db_structure.xml and belongs to dav app.
Does this mean that I should rewrite entire core db structure to Migrations?
I think I can't use Migrations just for one table. So the choice either move the table to proper app (dav) and put `use_migrations` into its `info.xml` or do the same for the entire core.

Comment (PVince81) [inline apps/dav/appinfo/Migrations/Version20170116170538.php]: A migration should be possible for just a single table.

Note that when enabling the app the first time, ownCloud will still setup the DB based on db_structure.xml. But updates have to go through migrations.

So if you setup that one table within the migration (and move it out of db_structure.xml), it should work.

**Comment by VicDeo** [comment]
Closed in favor of https://github.com/owncloud/core/pull/27005

**Pull request closed without merging**